src/main/java/net/sf/jabref/bibtex/DuplicateCheck.java

```diff
@@ -234,88 +240,59 @@ static double correlateByWords(String s1, String s2, boolean truncate) {
         return 1 - missRate;
     }
 
-    private static double correlateStrings(String s1, String s2, boolean truncate) {
-        int minLength = Math.min(s1.length(), s2.length());
-        if (truncate && (minLength == 1)) {
-            return s1.charAt(0) == s2.charAt(0) ? 1.0 : 0.0;
-        } else if ((s1.length() == 1) && (s2.length() == 1)) {
-            return s1.equals(s2) ? 1.0 : 0.0;
-        } else if (minLength == 0) {
-            return s1.isEmpty() && s2.isEmpty() ? 1.0 : 0;
-        }
 
-        // Convert strings to numbers and harmonize length in a method dependent on truncate:
-        if (truncate) {
-            // Harmonize length by truncation:
-            if (s1.length() > minLength) {
-                s1 = s1.substring(0, minLength);
-            }
-            if (s2.length() > minLength) {
-                s2 = s2.substring(0, minLength);
-            }
-        }
-        double[] n1 = DuplicateCheck.numberizeString(s1);
-        double[] n2 = DuplicateCheck.numberizeString(s2);
-        // If truncation is disabled, harmonize length by interpolation:
-        if (!truncate) {
-            if (n1.length < n2.length) {
-                n1 = DuplicateCheck.stretchArray(n1, n2.length);
-            } else if (n2.length < n1.length) {
-                n2 = DuplicateCheck.stretchArray(n2, n1.length);
-            }
+    /**
+     * Calculates the similarity (a number within 0 and 1) between two strings.
+     * http://stackoverflow.com/questions/955110/similarity-string-comparison-in-java
+     */
+    private static double similarity(String s1, String s2) {
+        String longer = s1, shorter = s2;
+        if (s1.length() < s2.length()) { // longer should always have greater length
+            longer = s2;
+            shorter = s1;
         }
-        return DuplicateCheck.corrCoef(n1, n2);
-    }
+        int longerLength = longer.length();
+        if (longerLength == 0) {
+            return 1.0;
```

Comment:
Correct, I haven't added any tests, just enabled the old ones. There are probably quite a bit more things to test, like complete entries, to get complete code coverage.

I can do that, but somehow, the interesting thing would be to see how it behaves in practice.

(Those lines would just return that the strings are identical in case both are `""`.)